src/ibex_bluesky_core/plans/reflectometry/det_map_align.py

```diff
@@ -0,0 +1,241 @@
+
+
+logger = logging.getLogger(__name__)
+
+
+def _height_scan_callback_and_fit(
+    reducer: PeriodSpecIntegralsReducer, height: NamedMovable[float], ax: Axes
+) -> tuple[DetMapHeightScanLiveDispatcher, LiveFit]:
+    intensity = "intensity"
+    height_scan_ld = DetMapHeightScanLiveDispatcher(
+        mon_name=reducer.mon_integrals.name,
+        det_name=reducer.det_integrals.name,
+        out_name=intensity,
+    )
+
+    height_scan_callbacks = ISISCallbacks(
+        x=height.name,
+        y=intensity,
+        yerr=f"{intensity}_err",
+        fit=Gaussian().fit(),
+        add_peak_stats=False,
+        ax=ax,
+        live_fit_logger_postfix="_height",
+        human_readable_file_postfix="_height",
+    )
+    for cb in height_scan_callbacks.subs:
+        height_scan_ld.subscribe(cb)
+
+    return height_scan_ld, height_scan_callbacks.live_fit
+
+
+def _angle_scan_callback_and_fit(
+    reducer: PeriodSpecIntegralsReducer, angle_map: npt.NDArray[np.float64], ax: Axes
+) -> tuple[DetMapAngleScanLiveDispatcher, LiveFit]:
+    angle_name = "angle"
+    counts_name = "counts"
+
+    angle_scan_ld = DetMapAngleScanLiveDispatcher(
+        x_data=angle_map,
+        x_name=angle_name,
+        y_in_name=reducer.det_integrals.name,
+        y_out_name=counts_name,
+    )
+
+    angle_scan_callbacks = ISISCallbacks(
+        x=angle_name,
+        y=counts_name,
+        yerr=f"{counts_name}_err",
+        fit=Gaussian().fit(),
+        add_peak_stats=False,
+        add_table_cb=False,
+        ax=ax,
+        live_fit_logger_postfix="_angle",
+        human_readable_file_postfix="_angle",
+    )
+    for cb in angle_scan_callbacks.subs:
+        angle_scan_ld.subscribe(cb)
+
+    return angle_scan_ld, angle_scan_callbacks.live_fit
+
+
+def _check_angle_map_shape(
+    reducer: PeriodSpecIntegralsReducer,
+    angle_map: npt.NDArray[np.float64],
+) -> None:
+    if reducer.detectors.shape != angle_map.shape:
+        raise ValueError(
+            f"detectors ({reducer.detectors.shape}) and "
+            f"angle_map ({angle_map.shape}) must have same shape"
+        )
+
+
+def angle_scan_plan(
+    dae: SimpleDae[PeriodPerPointController, Waiter, PeriodSpecIntegralsReducer],
+    *,
+    angle_map: npt.NDArray[np.float64],
+) -> Generator[Msg, None, ModelResult | None]:
+    """Reflectometry detector-mapping angle alignment plan.
+
+    This plan does not move any axes, but takes a single DAE measurement at the current
+    position. It processes this single DAE measurement to produce an angle scan, mapping
+    the detector intensity against angular pixel position.
+
+    Args:
+        dae: The DAE to acquire from
+        angle_map: a numpy array, with the same shape as detectors,
+            describing the detector angle of each detector pixel
+
+    """
+    logger.info("Starting angle scan")
+    reducer = dae.reducer
+    _check_angle_map_shape(reducer, angle_map)
```

Comment:
is it worth adding a generic helper that checks the controller is a runperperiod controller and reducer is a PeriodSpecIntegralsReducer

we might want to use this in plans generally?